action/protocol/staking/handlers.go

```diff
@@ -426,6 +430,10 @@ func (p *Protocol) handleRestake(ctx context.Context, act *action.Restake, csm C
 	prevWeightedVotes := p.calculateVoteWeight(bucket, csm.ContainsSelfStakingBucket(act.BucketIndex()))
 	// update bucket
 	bucket.StakedDuration = time.Duration(act.Duration()) * 24 * time.Hour
```

Comment:
in case of switching off autostake, stakedDuration should be same as before

otherwise, user can create autostake bucket for duration=30 days, switch it off with duration = 7 days, then i can withdraw after 7 + 14 days, while we expect user need to wait 30 + 14 days before withdraw

Reply:
we changed implementation that when user withdraw, they only wait for 14days. But after switching off autostake, user have to wait for duration and be able to unstake after it.

Reply:
Also for the changing duration, from what I understand, duration is not to make user to hold for the duration but to affect voting power, because we designed to update for user to restake and update duration "anytime". This issue is not only applied to this case which is updating autostake.  

For example, we can think that if user autoStake = false, he set up 10 days for duration. So we expect that he has to wait 10 days to unstake, but he can restake to change it to 5 days, and he will be able to unstake it after 5 days, but this is how we designed. Changing duration can affect his voting power tho.

Reply:
i see, thx for the explanation

my question is: user can stake with duration = 3 month, autostake = ON, then enjoy a higher reward %
after 3 month, he restake to duration = 7 days, autostake = OFF, then he only need to wait 7 days to unstake, then he can withdraw (wait another 14 days)

while our design for autostake should be: to enjoy the higher reward, you need to wait the original duration at the time of unstake, that is, anytime he wanted to unstake, he need to wait 3 months from the time of unstake

but now if we change that to 7 days, he only needs to wait 7 days